### ISSUE TITLE
Error fetching the genesis block from the network

### ISSUE THREAD

**Issue by econcotelli** [input-output-hk/jormungandr#1132]
**Describe the bug**

**Recurrent message from all bootstrap trusted peers:**

Nov 14 14:18:37.390 INFO fetching block cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, peer_address: 18.196.168.220:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:18:38.491 WARN failed to download block, error: GetBlocksStream { source: Error { code: InvalidArgument, source: Custom { kind: InvalidInput, error: StructureInvalid("fragment padding tag expected at 0 but got 2") } } }, peer_address: 18.196.168.220:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init

**Mandatory Information**

"jcli 0.7.0 (jorm_v0.7.0-a93d4f67, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]"

"jormungandr 0.7.0 (jorm_v0.7.0-a93d4f67, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]"

**To Reproduce**

Steps to reproduce the behavior: **Run Jormungandr**

[nix-shell:~/jormungandr]$ launch_jormungandr

Nov 14 14:34:08.928 INFO Starting jormungandr 0.7.0 (jorm_v0.7.0-a93d4f67, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)], task: init
Nov 14 14:34:08.928 INFO storing blockchain in '"/home/econcotelli/JormungandrWorkDir/Blockchain/blocks.sqlite"', task: init
Nov 14 14:34:08.934 INFO fetching block cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, peer_address: 184.169.162.15:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:09.361 WARN failed to download block, error: GetBlocksStream { source: Error { code: InvalidArgument, source: Custom { kind: InvalidInput, error: StructureInvalid("fragment padding tag expected at 0 but got 2") } } }, peer_address: 184.169.162.15:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:09.361 INFO fetching block cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, peer_address: 13.56.87.134:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:09.723 WARN failed to download block, error: GetBlocksStream { source: Error { code: InvalidArgument, source: Custom { kind: InvalidInput, error: StructureInvalid("fragment padding tag expected at 0 but got 2") } } }, peer_address: 13.56.87.134:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:09.723 INFO fetching block cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, peer_address: 13.230.137.72:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:10.263 WARN failed to download block, error: GetBlocksStream { source: Error { code: InvalidArgument, source: Custom { kind: InvalidInput, error: StructureInvalid("fragment padding tag expected at 0 but got 2") } } }, peer_address: 13.230.137.72:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:10.264 INFO fetching block cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, peer_address: 18.196.168.220:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:10.337 WARN failed to download block, error: GetBlocksStream { source: Error { code: InvalidArgument, source: Custom { kind: InvalidInput, error: StructureInvalid("fragment padding tag expected at 0 but got 2") } } }, peer_address: 18.196.168.220:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:10.337 INFO fetching block cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, peer_address: 3.124.132.123:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:10.414 WARN failed to download block, error: GetBlocksStream { source: Error { code: InvalidArgument, source: Custom { kind: InvalidInput, error: StructureInvalid("fragment padding tag expected at 0 but got 2") } } }, peer_address: 3.124.132.123:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:10.414 INFO fetching block cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, peer_address: 18.184.181.30:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:10.489 WARN failed to download block, error: GetBlocksStream { source: Error { code: InvalidArgument, source: Custom { kind: InvalidInput, error: StructureInvalid("fragment padding tag expected at 0 but got 2") } } }, peer_address: 18.184.181.30:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:10.489 INFO fetching block cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, peer_address: 13.230.48.191:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Nov 14 14:34:11.058 WARN failed to download block, error: GetBlocksStream { source: Error { code: InvalidArgument, source: Custom { kind: InvalidInput, error: StructureInvalid("fragment padding tag expected at 0 but got 2") } } }, peer_address: 13.230.48.191:3000, block: cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58, task: init
Error fetching the genesis block from the network
 |-> could not download block hash cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58

**Expected behavior**
Get the node running fine.

**Additional context**

Hello IOHK Dev Team,

I'm getting above errors running v0.7.0, using last configuration taken from https://hydra.iohk.io/build/1334642.

I've just updated new trusted peers in my node-config.yaml file.
As per mentioned configuration link I noticed that genesis block hash is still the one used in RC7 and some previous ones -> cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58

I made a try by reinstalling previous RC7 version, and it works fine with new trusted peers for v0.7.0. 

I'm on
@nixos:~]$ nixos-version 
19.09.1223.cb2cdab7136 (Loris)

Script "launch_jormungandr" executes:

jormungandr \
--genesis-block-hash cfd99bc54ebf44b44e72db7e2d48a40499888781e7628ea0fbf286bfd480ca58 \
--config $conf/node-config-R7.yaml \
--secret $secr/pool-secret.yaml

Please let me know if further information are needed,
many thanks in advance,
kind regards,
Emanuele

**Comment by econcotelli** [comment]
Hello Nicolas,
I was connecting to nightly, many thanks for your feedback, I'm going to apply beta config
"https://hydra.iohk.io/build/1298124"
Regards,
Emanuele

**Comment by NicolasDP** [comment]
Thanks. `nightly` is not compatible (yet). and `beta` is not yet too. Though I have heard it will happen soon.

**Comment by econcotelli** [comment]
Hello Nicolas,
so if both nightly and beta configs are not usable on v0.7.0, should I bootstrap in private mode,
by creating a genesis using bootstrap -g -s 2?

In this case I'm getting a new error for which bootstrap script needs <STAKE_KEY> and <STAKE_POOL_IDS> values. I tried to pass them (my stakekey and pool.id)
as Int value and also via $(cat file) but still same error:

########################################################

* Consensus      : genesis_praos
* REST Port      : 8443
* Slot duration  : 2
* Slots per epoch: 5000

########################################################

* CLI  version: jcli 0.7.0
* NODE version: jormungandr 0.7.0

########################################################

error: Invalid value for '<STAKE_KEY>': Failed to parse bech32, invalid data format
Invalid certificate
  |-> expected certificate in bech32
  |-> invalid length
error: Invalid value for '<STAKE_POOL_IDS>...': invalid hex encoding for hash value
Invalid certificate
  |-> expected certificate in bech32
  |-> invalid length
genesis file corrupted
  |-> initial: invalid length at line 18 column 3
error: building the genesis block

Many thanks
best regards.
Emanuele

**Comment by econcotelli** [comment]
Sorry, I had a look and the bootstrap script doesn't requires those argument.

**Comment by frasermince** [comment]
Hey @econcotelli! I'm running into the same problem running in private getting the invalid values for '<STAKE_KEY>' and '<STAKE_POOL_IDS>...' on the bootstrap script. What is the solution?

**Comment by econcotelli** [comment]
Hello @frasermince, 

not solved on my side, I made a diff between rc7_bootstrap and v0.7.0_bootscripts, and new version has changes. Bootstrap of rc7 works. 

I let IOHK developers provide a feedback on this. If is possible.

Meanwhile, good news, new configurations for: beta, nighlty ad qa*
for v0.7.0 are available on https://hydra.iohk.io/eval/774488#tabs-still-succeed
I've just tested nightly and qa, and both run fine.

*qa: I don't know what qa indicates, but there is repo in IOHK named jormungandr-qa which contains several scripts.

regards,
Emanuele

**Comment by rdlrt** [comment]
Guys github is not a chat/support platform, everyone subscribed gets a notification for every message you type. Can we move this to [workgroup](https://t.me/CardanoStakePoolWorkgroup) ?

**Comment by econcotelli** [comment]
Hello Rdirt,
I deduced that this is not the right place for certain activities. Sorry.
Telegram workgroup for stake pool operator, is what I need, more than rest of cardano telegram groups.
I'm currently able to enter and join to cardano groups, except for that one, for which I get message "chat is not available".
May  I've  been banned/blocked, but is quite strange considering that a I made no post there, when I joined about a week ago.
Should I ask someone for authorization to access mentioned group?
Many thanks in advance
best regards
Emanuele
